core/trino-main/src/main/java/io/trino/sql/planner/TypeAnalyzer.java

```diff
@@ -39,23 +43,25 @@
  */
 public class TypeAnalyzer
 {
-    private final SqlParser parser;
-    private final Metadata metadata;
+    private final PlannerContext plannerContext;
+    private final StatementAnalyzerFactory statementAnalyzerFactory;
+    private final AccessControl accessControl;
 
     @Inject
-    public TypeAnalyzer(SqlParser parser, Metadata metadata)
+    public TypeAnalyzer(PlannerContext plannerContext, StatementAnalyzerFactory statementAnalyzerFactory, AccessControl accessControl)
```

Comment:
Previous TypeAnalyzer signature was: `TypeAnalyzer(SqlParser parser, Metadata metadata)`.
That was kind of easy to understand.
Why does it need all these things now?
@martint can you perhaps explain this to me?

Reply:
In particular, in a unit test i could create TypeAnalyzer like this

```java
TypeAnalyzer typeAnalyzer = new TypeAnalyzer(new SqlParser(), createTestMetadataManager());
```

Now it seems this takes
```
Metadata metadata = createTestMetadataManager();
TypeOperators typeOperators = new TypeOperators();
InternalTypeManager typeManager = new InternalTypeManager(new TypeRegistry(typeOperators, new FeaturesConfig()));
PlannerContext plannerContext = new PlannerContext(
        metadata,
        typeOperators,
        new InternalBlockEncodingSerde(new BlockEncodingManager(), typeManager),
        TESTING_TYPE_MANAGER);
TypeAnalyzer typeAnalyzer = createTestingTypeAnalyzer(plannerContext);
```

to get same thing.